python/pyspark/sql/functions.py

```diff
@@ -4703,7 +4741,8 @@ def array_repeat(col: "ColumnOrName", count: Union[Column, int]) -> Column:
     assert sc is not None and sc._jvm is not None
     return Column(
         sc._jvm.functions.array_repeat(
-            _to_java_column(col), _to_java_column(count) if isinstance(count, Column) else count
+            _to_java_column(col),
+            count if isinstance(count, int) else _to_java_column(count),
```

Comment:
I'd prefer to either leave this one for the time being, or use the same approach as for the previous PR (`lit`).